packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -784,7 +787,12 @@ export class CdkToolkit {
       defaultBehavior: DefaultSelection.OnlySingle,
     });
 
-    // No validation
+    const notExistPatterns = selector.patterns.filter(pattern => !stacks.stackArtifacts.find(stack =>
+      minimatch(stack.hierarchicalId, pattern) || (stack.id === pattern && semver.major(versionNumber()) < 2),
+    ));
+    if (notExistPatterns.length > 0) {
+      warning(`${notExistPatterns.join(', ')} ${notExistPatterns.length === 1 ? 'does' : 'do'} not exist.`);
```

Comment:
The wording in this warning doesn't really tell the user why the stack does not exist. If it's the case that we're just checking for case insensitive matches then maybe we can explicitly tell the user that's the case or point out that they need to change their capitalization. Using the `iamstack` vs `IAMStack` example, something like:

 ```
iamstack does not exist. Do you mean IAMStack?
``` 
or 
```
iamstack does not exist but closely matches IAMStack (or simply another stack name). Please check your casing.
```